### DISCUSSION TITLE
How to OTA update?

### DISCUSSION THREAD

**Discussion by greymda** [hoylabs/OpenDTU-OnBattery#1148] [Q&A] [answered]
Stupid question, but still.
I want to update my 4mb board to the latest version.
What image should i chose to preserve my configuration and pin settings?
To be honest a bit confused and and i cannot find any such page in Wiki (i think community would benefit from such page)

![IMG_6986](https://github.com/user-attachments/assets/28081993-6268-44aa-8f4a-ce2ac0abe7c9)

**Comment by spcqike** [comment, marked as answer]
Should be plain generic. The first one. 

In doubt check your opendtus info page. You probably need the same PIO environment 

in my case s3_usb 

<img width="528" alt="image" src="https://github.com/user-attachments/assets/c82186e8-9270-403b-bd49-9bd16f00c685">

**Comment by greymda** [comment]
this is mine:
<img width="797" alt="Screenshot 2024-08-02 at 2 52 12 PM" src="https://github.com/user-attachments/assets/19808a86-99d8-45a8-bd8f-e115dbfd9733">

**Reply by greymda** [reply]
@spcqike  @schlimmchen  am i correct assuming that in my case (see the picture above) I must update with  opendtu-onbattery-generic_esp32.bin  and all my settings will be preserved?

**Reply by spcqike** [reply]
AFAIK generic and generic_esp32 are the same. So yes. 

In doubt download your config file and pinmapping beforehand.

**Reply by greymda** [reply]
thanks, having a backup is always a good idea indeed.

**Reply by schlimmchen** [reply]
> am i correct assuming that in my case (see the picture above) I must update with opendtu-onbattery-generic_esp32.bin and all my settings will be preserved?

Yes.

> AFAIK generic and generic_esp32 are the same.

Except for some default pin mappings, which nobody should rely on (use a working and full pin_mapping.json instead).

**Reply by greymda** [reply]
reporting back. updated with the onbattery-generic_esp32.bin -  all the settings and stats are preserved. thanks and happy friday (also international beer day!)
